app/Lib/Tools/BackgroundJobsTool.php

```diff
@@ -0,0 +1,647 @@
+     * @return string Job Id.
+     */
+    private function resqueEnqueue(
+        string $queue,
+        string $class,
+        $args = [],
+        $trackStatus = null,
+        int $jobId = null
+    ): string {
+
+        $process_id = CakeResque::enqueue(
+            $queue,
+            $class,
+            $args,
+            $trackStatus
+        );
+
+        if ($jobId) {
+            $this->updateJobProcessId($jobId, $process_id);
+        }
+
+        return $process_id;
+    }
+
+    /**
+     *  Dequeue a BackgroundJob.
+     *  If the queue is empty the read is blocked until a job is pushed to this queue or the timeout is reached.
+     *
+     * @param string $queue Queue name, e.g. 'default'.
+     * @param int    $timeout Time to block the read if the queue is empty.
+     *                  Must be less than your configured `read_write_timeout` 
+     *                  for the redis connection.
+     * 
+     * @return BackgroundJob|null.
+     * @throws Exception
+     */
+    public function dequeue($queue, int $timeout = 30): ?BackgroundJob
+    {
+        $this->validateQueue($queue);
+
+        $rawJob = $this->RedisConnection->blpop($queue, $timeout);
+
+        if (!empty($rawJob)) {
+            try {
+                return new BackgroundJob($rawJob[1]);
+            } catch (Exception $exception) {
+                CakeLog::error("Failed to parse job, invalid format: {$rawJob[1]}. exception: {$exception->getMessage()}");
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Get the job status.
+     *
+     * @param string $jobId Backgroun Job Id.
+     * 
+     * @return BackgroundJob|null job status.
+     * 
+     */
+    public function getJob(string $jobId): ?BackgroundJob
+    {
+        $rawJob = $this->RedisConnection->get(
+            self::JOB_STATUS_PREFIX . ':' . $jobId
+        );
+
+        if ($rawJob) {
+            return new BackgroundJob($rawJob);
+        }
+
+        return null;
+    }
+
+    /**
+     * Get the queues's names.
+     *
+     * @return array Array containing the queues' names.
+     */
+    public function getQueues(): array
+    {
+        return self::VALID_QUEUES;
+    }
+
+    /**
+     * Clear all the queue's jobs.
+     *
+     * @param string $queue Queue name, e.g. 'default'.
+     * 
+     * @return boolean True on success, false on failure.
+     */
+    public function clearQueue($queue): bool
+    {
+        $this->validateQueue($queue);
+
+        return (bool) $this->RedisConnection->del($queue);
+    }
+
+    /**
+     * Get all workers' instances.
+     *
+     * @return Worker[] List of worker's instances.
+     */
+    public function getWorkers(): array
+    {
+        $workers = [];
+        $procs = $this->Supervisor->getAllProcesses();
+
+        foreach ($procs as $proc) {
+            if ($proc->offsetGet('group') === self::MISP_WORKERS_PROCESS_GROUP) {
```

Comment:
Check if it's running - if not then pid returned is 0 and the $proc object contains the reasons for the failure.

Reply:
I'll skip the worker if pid=0